File: Alfombra.rb
class Alfombra

	base = "&&&\n& &\n&&&"
	p = "&&&"
	s = "& &"
	t = "&&&"
	e = " "
	##NIVEL 0
	puts "&&&\n&&&\n&&&"
	puts
	##NIVEL 1
	puts base
	puts
	
	##NIVEL 2
		for j in 1..3
			if j == 2 then
				puts p+(e*3)+p, s+(e*3)+s, t+(e*3)+t
				next
			end
			puts p*3, s*3, t*3
		end
		puts

	##NIVEL 3
		for j in 1..9
			if j == 5 then
				puts (p+(e*3)+p)+(e*9)+(p)+(e*3)+(p), (s+(e*3)+s)+(e*9)+(s+(e*3)+s), (t+(e*3)+t)+(e*9)+(t+(e*3)+t)
				next
			end
			if j == 2 || j == 8 then
				puts (p+(e*3)+p)*3, (s+(e*3)+s)*3, (t+(e*3)+t)*3
				next
			end
			puts p*9, s*9, t*9
		end
		puts
	
	#for j in 1..27
	#	if j == 14 then
	#		puts (p*9)+(e*27)+(p*9), (s*9)+(e*27)+(s*9), (t*9)+(e*27)+(t*9)
	#		next
	#	end
	#	puts p*27, s*27, t*27
	#end
end

File: alfombra.jl
#=
  program:1
  Name: alfombra de sierpinski.cpp
  Author: josue acevedo maldonado, maldad, Luis E. Ramos Cruz
  Date: 27/08/15 
  Description:
=#


I=0
F=27
cn=1
puntos=[]
push!(puntos,I)
push!(puntos,F)

function dividir(pi, pf)
	if pi<pf
		t= (pf-pi)/3
		pc1= pi+t
		pc2= pc1+t
				#=print ("\033[%d;%dH",cn,40)
				print("pc1: %d\n",pc1 );=#
		push!(puntos,pc1)
				#=print ("\033[%d;%dH",cn,40)
				print("pc2: %d\n",pc2 );=#
		push!(puntos,pc2)	
	end
end


function pintar(al,pi,pf,le)

print("\033[0;0H")
print("\033[$(al+1);0H")
	for i=pi:(pf-1)
		print("\033[$(pi+1)C")
		for j = pi:(pf-1)
    		print(le)
    	end
    	print("\n")
    end

end



    x="\033[1;37m.\033[1;32m"
    O="▓" # ▆ ▇ ▉ ⬣ ◼ ▉ 

    print("Alfombra de Sierpinski \n")

    print("ingrese el nivel (minimo 1, maximo 3):\n")
 
    print("\033[1J"); #BORRA LA TERMINAL
    #print ("\033[47m"); #color de fondo blanco
    print("\033[1;32m"); 
    pintar(I,I,F,x)


for k=1:3
	ni=length(puntos)
	for i =1:length(puntos)
    	#=print ("\033[%d;%dH",cn,60)
    	    	print("%d .- primer punto %d\n",cn,puntos[i])
    	    	print ("\033[%d;%dH",cn,80)
    	    	print("segundo punto %d\n", puntos[i+1]);=#
    	dividir(puntos[i],puntos[i+1])
    end

	
	ni+=1

	for i = ni:2:length(puntos), j = ni:2:length(puntos)
		pintar(puntos[i],puntos[j],puntos[j+1],O)
	end
	
	sort(puntos)
end
	
	print("\033[1;37m")
    print("\033[$(F+2);1H")

 
   
